src/main.c

```diff
@@ -42,7 +48,31 @@ int main(int argc, char **argv)
 	while (true) {
 		volatile exception_s e;
 		TRY_CATCH (e, EXCEPTION_ALL) {
-			gdb_main();
+			SET_IDLE_STATE(false);
+			while (gdb_target_running && cur_target) {
+				gdb_poll_target();
+
+				// Check again, as `gdb_poll_target()` may
+				// alter these variables.
+				if (!gdb_target_running || !cur_target) {
+					break;
+				}
+				char c = (char)gdb_if_getchar_to(0);
```

Comment:
This cast should be removed - it should do nothing now and was a bug that it was ever required.